File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/MeaningOfLife.scala
package com.xebia.functional.xef.scala.conversation

import com.xebia.functional.xef.scala.conversation.*
import io.circe.Decoder
import com.xebia.functional.xef.prompt.Prompt

private final case class MeaningOfLife(mainTheories: List[String]) derives SerialDescriptor, Decoder

@main def runMeaningOfLife: Unit =
  conversation {
    val meaningOfLife = prompt[MeaningOfLife](Prompt("What are the main theories about the meaning of life"))
    println(s"There are several theories about the meaning of life:\n ${meaningOfLife.mainTheories}")
  }

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/MeaningOfLife.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.OpenAI
import com.xebia.functional.xef.auto.llm.openai.prompt
import kotlinx.serialization.Serializable

@Serializable data class MeaningOfLife(val mainTheories: List<String>)

suspend fun main() {
  OpenAI.conversation {
    val meaningOfLife: MeaningOfLife =
      prompt("What are the main theories about the meaning of life")
    println("There are several theories about the meaning of life:\n ${meaningOfLife.mainTheories}")
  }
}
